cmd/bedPeOverlap/bedPeOverlap.go

```diff
@@ -125,6 +127,51 @@ func SelectIsBed(bedSelectFile string, bedpeInFile string, overlapThreshold floa
 	exception.PanicOnErr(err)
 }
 
+func SelectIsBedBoth(bedSelectFile string, bedpeInFile string, overlapThreshold float64, contactOutFile string) {
+	var selectIntervals = make([]interval.Interval, 0)
+	var currOverlaps []interval.Interval
+	var bothOverlap []interval.Interval
+	var err error
+	var found bool
+
+	selectRecords := bed.Read(bedSelectFile)
+	inBedPe := bedpe.Read(bedpeInFile)
+	out := fileio.EasyCreate(contactOutFile)
+
+	for _, i := range selectRecords {
+		selectIntervals = append(selectIntervals, i)
+	}
+	selectTree := interval.BuildTree(selectIntervals)
+
+	for _, i := range inBedPe {
+		currOverlaps = interval.Query(selectTree, i.A, "any")
+		if len(currOverlaps) > 0 { //if A, the left side of the input bedPe, overlaps any of the select beds, write the bedPe to output.
+			if overlapThreshold == 0 {
+				bothOverlap = interval.Query(selectTree, i.B, "any")
+				if len(bothOverlap) > 0 {
+					bedpe.WriteToFileHandle(out, i)
+				}
+			} else {
+				found = false
+				for _, j := range currOverlaps {
```

Comment:
I would pull the i,j,k loop iterator variable declarations up top to cut down on allocations